testplan/monitor/resource.py

```diff
@@ -0,0 +1,339 @@
+import os
+import csv
+import time
+import json
+import socket
+import pathlib
+import asyncio
+import logging
+import dataclasses
+
+import zmq
+import zmq.asyncio
+import zmq.sugar.context
+import psutil
+import multiprocessing
+
+from typing import Dict, Optional, Union, TextIO
+from testplan.common.utils.strings import slugify
+from testplan.common.utils.logger import LOGFILE_FORMAT
+from testplan.common.utils.timing import wait
+from testplan.runners.pools import communication
+from testplan.common.serialization.base import serialize, deserialize
+
+
+@dataclasses.dataclass
+class ResourceData:
+    cpu_usage: float
+    memory_used: int
+    iops: float
+    io_read: float
+    io_write: float
+
+
+@dataclasses.dataclass
+class HostResourceData(ResourceData):
+    disk_used: int
+
+
+@dataclasses.dataclass
+class ProcessResourceData(ResourceData):
+    name: str
+    cmdline: str
+    pid: int
+
+
+class ResourceMonitorClient:
+    def __init__(
+        self,
+        collector_address: str,
+        parent_pid: int = os.getpid(),
+        host: Optional[str] = None,
+        cpu_count: Optional[int] = None,
+        memory_size: int = 0,
+        disk_path: str = "/",
```

Comment:
keep disk_path, and default to pwd() (since client does not have a runpath, otherwise runpath will be a better option)

remove all other parameters